src/index.ts

```diff
@@ -9,65 +10,145 @@ class VPCPlugin {
+    };
   }
 
   /**
    * Wrapper for lifecycle function, initializes variables and checks if enabled.
    * @param lifecycleFunc lifecycle function that actually does desired action
    */
   public async hookWrapper (lifecycleFunc: any) {
-    this.validateConfigExists()
-    this.initAWSResources()
-    return await lifecycleFunc.call(this)
+    this.validateConfigExists();
+    this.initAWSResources();
+    return await lifecycleFunc.call(this);
   }
 
   /**
    * Validate if the plugin config exists
    */
   public validateConfigExists (): void {
-    const config = this.serverless.service.custom
-    const vpc = config && config.vpc
+    const config = this.serverless.service.custom;
+    if (!config.vpcDiscovery) {
+      config.vpcDiscovery = config.vpc;
+      Globals.logWarning(
+        "The `vpc` option of `custom` config is deprecated and will be removed in the future. " +
+        "Please use `vpcDiscovery` option instead."
+      );
+    }
+    const vpc = config && config.vpcDiscovery;
     // Checks if the serverless file is setup correctly
-    if (!vpc || vpc.vpcName == null || vpc.subnetNames == null || vpc.securityGroupNames == null) {
-      throw new Error("Serverless file is not configured correctly. Please see README for proper setup.")
+    if (!vpc || vpc.vpcName == null || (vpc.subnetNames == null && vpc.securityGroupNames == null)) {
+      throw new Error(
+        "Serverless file is not configured correctly. " +
+        "You must specify the vpcName and at least one of subnetNames or securityGroupNames. " +
+        "Please see README for proper setup."
+      );
     }
   }
 
   /**
    * Setup AWS resources
    */
   public initAWSResources (): void {
-    const credentials = this.serverless.providers.aws.getCredentials()
-    credentials.region = this.serverless.providers.aws.getRegion()
-    this.ec2Wrapper = new EC2Wrapper(credentials)
+    const credentials = this.serverless.providers.aws.getCredentials();
+    credentials.region = this.serverless.providers.aws.getRegion();
+    this.ec2Wrapper = new EC2Wrapper(credentials);
   }
 
   /**
-   * Gets the desired vpc with the designated subnets and security groups
-   * that were set in serverless config file
+   * Updates functions vpc config
    * @returns {Promise<object>}
    */
-  public async updateVpcConfig (): Promise<object> {
-    this.serverless.cli.log("Updating VPC config...")
+  public async updateFunctionsVpcConfig (): Promise<object> {
+    Globals.logInfo("Updating VPC config...");
+    const service = this.serverless.service;
+
+    // Sets the serverless's vpc config
+    if (service.functions) {
+      // get basic VPC config
+      const basicVPCDiscovery = await this.getVpcConfig(service.custom.vpcDiscovery);
+      // loop through the functions and update VPC config
+      for (const fName of Object.keys(service.functions)) {
+        const f = service.functions[fName];
+        const fVPCDiscovery = f.vpcDiscovery;
```

Comment:
Maybe `func_vpc_config` instead?

Reply:
Yep, but I also try to name `vpc` and `vpcDiscobery` to understand which kind of options it has
like `subnetIds` or `subnetNames`